### DISCUSSION TITLE
How do I disable signature help?

### DISCUSSION THREAD

**Discussion by idelice** [LazyVim/LazyVim#4624] [General]
I want to disable this keymapping specified here:

[Link to keymap](https://github.com/LazyVim/LazyVim/blob/13a4a84e3485a36e64055365665a45dc82b6bf71/lua/lazyvim/plugins/lsp/keymaps.lua#L24)

I've tried adding this to my `keymap.lua`, but instead of overriding it, it adds an additional keymap to signature help:

`vim.keymap.set("i", "<c-s>", "<cmd>lua vim.lsp.buf.signature_help()<CR>", { desc = "Signature help" })`

**Comment by dpetka2001** [comment]
The [docs](https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps) describe how to do that.

**Reply by idelice** [reply]
> The [docs](https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps) describe how to do that.

```lua
return {
  "neovim/nvim-lspconfig",
  opts = function(_, opts)
    local keys = require("lazyvim.plugins.lsp.keymaps").get()
    keys[#keys + 1] = { "<c-k>", false }

    opts.diagnostics = {
      virtual_text = {
        severity = "Error",
      },
    }
    opts.inlay_hints = {
      enabled = true,
      exclude = { "typescriptreact" }, -- filetypes for which you don't want to enable inlay hints
    }
  end,
}
```

I can still do signature help in insert mode even tho i disabled it according to the docs.

Am I missing something here?

**Reply by dotfrag** [reply]
Try this:

```lua
    keys[#keys + 1] = { "<c-k>", mode = "i", false }
```

**Reply by idelice** [reply]
> Try this:
> 
> ```lua
>     keys[#keys + 1] = { "<c-k>", mode = "i", false }
> ```

Thanks that did the trick!

**Reply by dpetka2001** [reply]
Yep, `mode` is needed in `keys`, otherwise it's assumed `mode = "n"` for `normal` mode mappings if not specified. And `<c-k>` is a `insert` mode mapping.
